cmd/kube-proxy/app/server.go

```diff
@@ -463,7 +487,12 @@ with the apiserver API to configure the proxy.`,
 			if err := opts.Validate(args); err != nil {
 				klog.Fatalf("failed validate: %v", err)
 			}
-			klog.Fatal(opts.Run())
+			err := opts.Run()
+			if err, ok := err.(*OSExit); ok {
+				// os exit with status 0
+				err.Run(0)
+			}
+			klog.Fatal(err)
```

Comment:
IIRC, `klog.Fatal` can accept nil here. Wondering if it's cleaner if we keep this as just:
```go
klog.Fatal(opts.Run())
```

and in `runLoop()`:
```go
func (o *Options) runLoop() error {
        if o.CleanupAndExit {
              return o.proxyServer.CleanUpAndExit()
        }
```

Actually exiting with 0 is nice here but maybe not worth the trade-off since previous versions have been exiting with non-zero exit codes already? Either way, might be good change the behavior for exit codes in a follow-up PR to make the cherry-pick to previous versions easier.

Reply:
@andrewsykim Yeah, but should it use ``klog.Info`` rather than ``klog.Fatal`` when clean up and exit?